controllers/openstackdataplanerole_controller.go

```diff
@@ -96,11 +97,60 @@ func (r *OpenStackDataPlaneRoleReconciler) SetupWithManager(mgr ctrl.Manager) er
 
 // ReconcileNodes ensure the desired state for the nodes
 func (r *OpenStackDataPlaneRoleReconciler) ReconcileNodes(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole) error {
-	// loop over r.Spec.DataPlaneNodes:
+	// loop over instance.Spec.DataPlaneNodes:
 	//   for each node:
 	//     (1) complete node.Spec based r.Spec.nodeTemplate, and values set on the
 	//         node (values on the node take precedence over those from the template)
 	//     (2) Create a CR of OpenStackDataPlaneNode from the node
 
+	for i, node := range instance.Spec.DataPlaneNodes {
+		OpenStackDataPlaneNodeName := fmt.Sprintf("%s-node-%d", instance.Name, i)
+		newNode := &dataplanev1beta1.OpenStackDataPlaneNode{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      OpenStackDataPlaneNodeName,
+				Namespace: instance.Namespace,
+			},
+		}
+		// Set new node with values from its entry on the DataPlaneNodes list
+		newNode.Spec.Role = instance.Name
+		newNode.Spec.Node.AnsibleHost = node.Node.AnsibleHost
```

Comment:
AnsibleHost might not be required, so we should probably add a check for it. When we generate the inventory, we use AnsibleHost if set, otherwise HostName.
And could also make HostName optional, and default to just Name.